### DISCUSSION TITLE
shelly 2.5 after upgrade windows covering

### DISCUSSION THREAD

**Discussion by StefanoGiuli** [community/community#54180] [Programming Help]
### Body

after upgrading to merlin shelly 2.5 windows covering with external push buttons. when i say open it closes and when i say close it opens. help me
this is the code:
 

{"c":{"io":[[[0],3],[[2],6],[[5,13],6,0,1],[[4,15],2]],"l":0,"b":[{"g":2,"t":5}]},"a":[{"t":45,"o":18,"c":14,"f":70,"0":{"r":[{"g":15},{"g":4,"v":1}]},"1":{"r":[{"g":15,"v":1},{"g":4}]},"2":{"r":[{"g":15},{"g":4}]},"3":{"r":[{"g":15},{"g":4,"i":0.2}]},"4":{"r":[{"g":15,"i":0.2},{"g":4}]},"f3":[{"g":5,"t":0}],"f4":[{"g":13,"t":0}]}]}
 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by RamtinKosari** [comment]
Hi dear @StefanoGiuli 
To troubleshoot the problem of the windows covering behaving opposite to the expected behavior when you say "open" or "close," you can consider the following steps:

Verify the wiring: Double-check the wiring connections between the Shelly 2.5 device and the external push buttons. Ensure that they are connected correctly and securely.

Check the configuration: Review the configuration settings for the Shelly 2.5 device. Make sure that the inputs and outputs are assigned correctly and correspond to the physical connections. Pay attention to the assignment of "open" and "close" commands to the appropriate inputs and outputs.

Test with manual control: Try manually operating the external push buttons and observe if the windows covering responds as expected. This step can help identify if the issue lies with the physical connections or the configuration.

Debug the code: If the issue persists, you may need to debug the code further. Consult the documentation or resources provided by the manufacturer of the Shelly 2.5 device or seek assistance from their support team. They can provide specific guidance on how to troubleshoot and resolve any configuration issues.

It's important to note that modifying the configuration of the Shelly 2.5 device should be done with caution and following the manufacturer's instructions. If you are unsure about making changes to the configuration, it's recommended to seek assistance from a professional or the manufacturer's support channels.

Here's an example code snippet that demonstrates how you can control the windows covering using external push buttons with the Shelly 2.5 device:

```c++
#include <Arduino.h>
#include <ShellySwitch.h>

// Define the pin numbers for the external push buttons
const int openButtonPin = 2;
const int closeButtonPin = 3;

// Create instances of the ShellySwitch class for controlling the outputs
ShellySwitch openSwitch(4);
ShellySwitch closeSwitch(5);

void setup() {
  // Set the pin modes for the push buttons
  pinMode(openButtonPin, INPUT_PULLUP);
  pinMode(closeButtonPin, INPUT_PULLUP);

  // Initialize the ShellySwitch instances
  openSwitch.begin();
  closeSwitch.begin();
}

void loop() {
  // Check if the open button is pressed
  if (digitalRead(openButtonPin) == LOW) {
    openSwitch.turnOn();
  } else {
    openSwitch.turnOff();
  }

  // Check if the close button is pressed
  if (digitalRead(closeButtonPin) == LOW) {
    closeSwitch.turnOn();
  } else {
    closeSwitch.turnOff();
  }

  // Update the ShellySwitch instances
  openSwitch.update();
  closeSwitch.update();
}
```
In this example, we use the ShellySwitch library, which provides a convenient interface for controlling outputs on the Shelly devices. Make sure to install the library before using this code.

In the setup() function, we set the pin modes for the external push buttons as INPUT_PULLUP, assuming they are connected with the common terminal connected to ground. We also initialize the ShellySwitch instances for controlling the open and close outputs.

In the loop() function, we continuously check the state of the push buttons. If the open button is pressed, the openSwitch is turned on using the turnOn() method. Similarly, if the close button is pressed, the closeSwitch is turned on. If the buttons are not pressed, the respective switches are turned off using the turnOff() method.

The update() method is called to update the state of the switches and control the corresponding outputs accordingly.

Please note that this code assumes a specific wiring configuration and uses the Arduino framework with the ShellySwitch library. You may need to adapt it based on your specific hardware, wiring, and programming environment.
